### PR TITLE
Add buttons that can open the file path directly.

### PR DESCRIPTION
**Pull request by CloudeaSoft**
[https://github.com/ppy/osu-tools/issues/201](url)
The codes for the issue#201.

### CODE REVIEW

File: PerformanceCalculatorGUI/Components/SettingsPopover.cs [hunk 1]
```diff
@@ -134,5 +171,20 @@ private void saveConfig()
...
+        }
+
+        private void openCacheDirectory()
+        {
+            openFolder(cacheBindable.Value);
+        }
+
+        private static void openFolder(string folderPath)
+        {
+            Process.Start("explorer.exe", folderPath);
```
Comment (stanriders) [inline PerformanceCalculatorGUI/Components/SettingsPopover.cs:187]: This wont work on platforms other than windows

File: PerformanceCalculatorGUI/Components/SettingsPopover.cs [hunk 2]
```diff
@@ -83,17 +84,53 @@ private void load(SettingsManager configManager, OsuConfigManager osuConfig)
...
+                                            Current = { BindTarget = pathBindable }
+                                        },
+                                        new RoundedButton
+                                        {
+                                            Anchor = Anchor.CentreLeft,
+                                            Origin = Anchor.CentreLeft,
+                                            Size = new Vector2(100, 40),
+                                            Margin = new MarginPadding { Left = 10 },
+                                            Text = "Open",
+                                            Action = openPathDirectory
```
Comment (stanriders) [inline PerformanceCalculatorGUI/Components/SettingsPopover.cs:110]: I'm not sure how useful this even is - you can always just copypaste the path from the text field into the file manager address bar

**Review by stanriders — CHANGES_REQUESTED** [review]

File: PerformanceCalculatorGUI/Components/SettingsPopover.cs (hunk 1 above)
Comment (CloudeaSoft) [reply PerformanceCalculatorGUI/Components/SettingsPopover.cs:187]: Ok, i'll try to make it more simply for other platforms.

File: PerformanceCalculatorGUI/Components/SettingsPopover.cs (hunk 2 above)
Comment (CloudeaSoft) [reply PerformanceCalculatorGUI/Components/SettingsPopover.cs:110]: I thought osu! have buttons for root path/skin path/beatmap path(in editor), so i tried to add this button.

File: PerformanceCalculatorGUI/Components/SettingsPopover.cs [hunk 3]
```diff
@@ -134,5 +173,39 @@ private void saveConfig()
...
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                // For Linux
+                Process.Start("xdg-open", path);
+            }
+            else
+            {
+                // For others, do nothing
+            }
+        }
```
Comment (smoogipoo) [inline PerformanceCalculatorGUI/Components/SettingsPopover.cs]: The correct solution here is to use `Storage.PresentExternally()`. You should be able to directly inject `GameHost` as a dependency here. Something like the following might be enough:
```
host.GetStorage(path).PresentExternally();
```

**Review by smoogipoo — CHANGES_REQUESTED** [review]

File: PerformanceCalculatorGUI/Components/SettingsPopover.cs (hunk 3 above)
Comment (CloudeaSoft) [reply PerformanceCalculatorGUI/Components/SettingsPopover.cs]: Thx for the advice! :)

File: PerformanceCalculatorGUI/Components/SettingsPopover.cs (hunk 2 above)
Comment (Ireadosucode) [reply PerformanceCalculatorGUI/Components/SettingsPopover.cs:116]: it's quality of life for sure- not to mention a lot of people don't know how to do that

Comment (Ireadosucode) [reply PerformanceCalculatorGUI/Components/SettingsPopover.cs:116]: somehow..

**Pull request closed without merging**